src/bootstrap/src/utils/tarball.rs

```diff
@@ -317,6 +317,9 @@ impl<'a> Tarball<'a> {
             channel::write_commit_hash_file(&self.overlay_dir, &info.sha);
             channel::write_commit_info_file(&self.overlay_dir, info);
         }
+
+        self.add_file(self.builder.config.config.as_ref().unwrap(), &self.overlay_dir, 0o644);
```

Comment:
`config.config` can be `None` if you don't specify a path and don't have a created `config.toml`. Probably won't happen on CI, but it shouldn't crash here. I would include the file only if is not `None`.